code/src/app/api/carInfo/route.ts

```diff
@@ -217,144 +166,105 @@ export async function GET(req: Request) {
  * @param {string} req.body.dateNextTireChange - The date of next tire change.
  * @returns {Promise<NextResponse>} A response object.
  * 200: Car edited successfully. Updated car info returned.
- * 400: Bad request. Duplicate license or Car ID not provided or invalid.
- * 404: Car ID not found in DB.
+ * 400: Car ID not provided or invalid.
+ * 401: Bad request. Duplicate license.
+ * 402: Need both new date and mileage for last maintenance.
+ * 403: Car ID not found in DB.
  * 500: Unexpected server error.
  * @throws {Error} Throws an error if there's an issue with the registration process.
  */
-export async function PUT(req: Request) {
-  const requestBody = await req.json();
-  const {
-    manufacturer,
-    type,
-    year,
-    license,
-    mileage,
-    model,
-    color,
-    seats,
-    condition,
-    mileageLastOilChange,
-    mileageLastTireChange,
-  } = requestBody;
-
-  let {
-    dateNextOilChange,
-    dateNextTireChange,
-    oilOverdue,
-    tireOverdue,
-    maintenanceRequired,
-  } = requestBody;
 
+export async function PUT(req: Request) {
   try {
-    // Get saved car object, validate car exist
-    const getRequest = await GET(req);
-    const { message, carObject } = await getRequest.json();
-    if (!carObject) {
-      return NextResponse.json({ message }, { status: getRequest.status });
-    }
+    const data = await req.json();
 
-    // Check existing license being registered with another car
-    const existingLicense = await car.findOne({
-      // eslint-disable-next-line no-underscore-dangle
-      $and: [{ license }, { _id: { $ne: carObject._id } }],
-    });
+    const entries = Object.entries(data);
+    const nonEmptyOrNull = entries.filter(
+      ([, val]) => val !== '' && val !== null,
+    );
+    const filteredData = Object.fromEntries(nonEmptyOrNull);
 
```

Comment:
In order to remove any properties from the (frontend) data  object that have values of an empty string or null.